### PR TITLE
Add serializer that serialize to stream

### PR DESCRIPTION
**Pull request by johevemi**
## References

* #0000

### CODE REVIEW

File: source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs [hunk 1]
```diff
@@ -116,5 +119,25 @@ public void Deserialize_JsonStringIsNull_ThrowsException(JsonSerializer sut)
         {
             Assert.Throws<ArgumentNullException>(() => sut.Deserialize<TestObject>(null!));
         }
+
+        [Theory]
+        [InlineAutoMoqData]
+        public async Task SerializeToStream(JsonSerializer sut, [Frozen] TestObject message)
```
Comment (BjarkeMeier) [inline source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs:125]: Why [Frozen]? I don't think it has any effect here.

File: source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs [hunk 2]
```diff
@@ -116,5 +119,25 @@ public void Deserialize_JsonStringIsNull_ThrowsException(JsonSerializer sut)
...
+        {
+            // Act
+            var stream = new MemoryStream();
+            await sut.SerializeAsync(stream, message);
+            var jsonFromStream = Encoding.UTF8.GetString(stream.ToArray());
```
Comment (BjarkeMeier) [inline source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs:130]: I believe these 3 lines are arrange and not act.

File: source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs [hunk 3]
```diff
@@ -116,5 +119,25 @@ public void Deserialize_JsonStringIsNull_ThrowsException(JsonSerializer sut)
...
+            var deserializedObject = sut.Deserialize<TestObject>(jsonFromStream);
```
Comment (BjarkeMeier) [inline source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs:131]: Only one of these lines are your act. And btw, the test name doesn't help clarify which. So you should probably also work with the test naming.

File: source/JsonSerialization/source/JsonSerialization/JsonSerializer.cs [hunk 4]
```diff
@@ -92,5 +92,15 @@ public string Serialize<TValue>(TValue value)
...
         }
+
+        public Task SerializeAsync<TValue>(Stream stream, TValue value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            return System.Text.Json.JsonSerializer.SerializeAsync<TValue>(stream, value, _options);
```
Comment (BjarkeMeier) [inline source/JsonSerialization/source/JsonSerialization/JsonSerializer.cs:103]: nit
```suggestion
            ArgumentNullException.ThrowIfNull(value);
            return JsonSerializer.SerializeAsync<TValue>(stream, value, _options);
```

File: source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs (hunk 1 above)
Comment (johevemi) [reply source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs]: Removed

File: source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs (hunk 3 above)
Comment (johevemi) [reply source/JsonSerialization/source/JsonSerialization.Tests/JsonSerializerTests.cs]: act on only  await sut.SerializeAsync(stream, message);

File: source/JsonSerialization/source/JsonSerialization/JsonSerializer.cs (hunk 4 above)
Comment (johevemi) [reply source/JsonSerialization/source/JsonSerialization/JsonSerializer.cs:103]: The other serializers are written this way, so I wont change this unless all the others are changed aswell

**Review by BjarkeMeier — APPROVED** [review]

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**